pom.xml

```diff
@@ -594,6 +594,32 @@
                 </executions>
             </plugin>
 
+            <!-- Scala style check -->
+            <plugin>
+                <groupId>org.scalastyle</groupId>
+                <artifactId>scalastyle-maven-plugin</artifactId>
+                <version>0.8.0</version>
+                <configuration>
+                    <verbose>false</verbose>
+                    <failOnViolation>false</failOnViolation>
+                    <failOnWarning>false</failOnWarning>
+                    <includeTestSourceDirectory>true</includeTestSourceDirectory>
+                    <sourceDirectory>${project.basedir}/src/main/scala</sourceDirectory>
+                    <testSourceDirectory>${project.basedir}/src/test/scala</testSourceDirectory>
+                    <detail>true</detail>
+                    <configLocation>${user.dir}/scalastyle_config.xml</configLocation>
```

Comment:
I am not aware of any practically relevant way to run `mvn ...` outside the project root, so `${user.dir}` seems to handle all situations correctly.